TypeInjections/Tool/Tool.fs

```diff
@@ -57,41 +64,52 @@ module Tool =
             checkoutCommit commit1 location pathOld
             checkoutCommit commit2 location pathNew
 
+            pathOld, pathNew, $"{path}/Output"
 
-        // -r [github repo url] [github commit hash] [github commit hash] [OutputPath]
-        elif flag.Contains("-r") then
-
-            Console.WriteLine("***** remote github")
-
-            if argvList.Length <> 5 then
-                failwith "usage: program -r GITHUB_URL OLD_COMMIT NEW_COMMIT OUTPUTFOLDER"
-
-            let url = argvList.Item(1)
+        else if not remote.IsEmpty then
+            Utils.log "***** remote github"
 
-            let commit1 = argvList.Item(2)
-            let commit2 = argvList.Item(3)
-
-            let path = argvList.Item(4)
+            let path = remote.Item(3)
             let pathOld = $"{path}/Old"
             let pathNew = $"{path}/New"
+            let url = remote.Item(0)
+            let commit1 = remote.Item(1)
+            let commit2 = remote.Item(2)
 
-            // lib2git requires an empty folder to place repos
             for a in [ pathOld; pathNew ] do
                 if Directory.Exists(a) then
                     Directory.Delete(a, true)
 
-            // grab repos of new and old projects
             checkoutCommit commit1 url pathOld
             checkoutCommit commit2 url pathNew
 
-        else
-            failwith "flag not found please try again"
 
+            pathOld, pathNew, $"{path}/Output"
+
+        else
+            failwith "unable to read flags"
```

Comment:
Do I understand correctly that `local`/`git`/`remote` are mutually exclusive, but at least one of them must be set, with the priority `local`>`git`>`remote`?

If so, please reflect that in the error message here. I don't think `failwith` is the right thing here. You want to print a message that the tool was called incorrectly and gracefully exit.

Reply:
Yes, you're correct

I thought this would be a failure, because none of the command line options have any arguments associated with them. We would only reach this `else` statement if none of the flags were correctly called. 

Should I put a help message help the user next time? It would be hard to catch the user's mistake at the point because this `else` statement means the user didn't enter any arguments for any option

Reply:
A way to avoid this problem is to use `command line verbs` instead of `command line options`. From the documentation, I think only one verb executes per run. 

Ex of verb:
`typecart local /Users/tschuerl/OldFolder /Users/tschuerl/NewFolder /Users/tschuerl/Output`
Ex of option:
`typecart --local /Users/tschuerl/OldFolder /Users/tschuerl/NewFolder /Users/tschuerl/Output`

Reply:
Yes, that would be nicer.

Reply:
And indeed, I think what we want is: 
`typecart local --old /Users/tschuerl/OldFolder --new /Users/tschuerl/NewFolder --out /Users/tschuerl/Output`

Then in code we should be able to refer to the arguments via these names, i.e. you have `local.[0]` or so now, but that should change to something like `local.old` maybe?

Reply:
Cool, I added a commit where we use CLI verbs instead of options